index.d.ts

```diff
@@ -146,11 +146,25 @@ declare namespace Dysnomia {
   type InteractionDataOptionsRole = InteractionDataOption<"ROLE", string>;
   type InteractionDataOptionsString = InteractionDataOption<"STRING", string>;
   type InteractionDataOptionsUser = InteractionDataOption<"USER", string>;
+  type InteractionDataOptionsAttachment = InteractionDataOption<"ATTACHMENT", string>;
   type InteractionDataOptionsWithOptions = InteractionDataOptionsSubCommand | InteractionDataOptionsSubCommandGroup;
-  type InteractionDataOptionsWithValue = InteractionDataOptionsString | InteractionDataOptionsInteger | InteractionDataOptionsBoolean | InteractionDataOptionsUser | InteractionDataOptionsChannel | InteractionDataOptionsRole | InteractionDataOptionsMentionable | InteractionDataOptionsNumber;
+  type InteractionDataOptionsWithValue = InteractionDataOptionsString | InteractionDataOptionsInteger | InteractionDataOptionsBoolean | InteractionDataOptionsUser | InteractionDataOptionsChannel | InteractionDataOptionsRole | InteractionDataOptionsMentionable | InteractionDataOptionsNumber | InteractionDataOptionsAttachment;
   type InteractionResponse = InteractionResponseAutocomplete | InteractionResponseDeferred | InteractionResponseLaunchActivity | InteractionResponseMessage | InteractionResponsePong | InteractionResponsePremiumRequired;
   type InteractionResponseTypes = Constants["InteractionResponseTypes"][keyof Constants["InteractionResponseTypes"]];
   type InteractionTypes = Constants["InteractionTypes"][keyof Constants["InteractionTypes"]];
+  interface InteractionDataOptionIdTypeMap {
```

Comment:
nit: We're using uppercase "ID" in the codebase, otherwise looks okay to me
```suggestion
  interface InteractionDataOptionIDTypeMap {
```

edit: actually, now that I think about it, the option types aren't really an ID per se, so probably naming it as a `type -> structure` mapping would be better?

Reply:
> the option types aren't really an ID per se

They are type-id literals. 

I think the technically correct name would be `InteractionDataOptionTypeIDTypeMap`, but I'll comply with any name you come up with.